scripts: Require Python 3.8

carlescufi opened the pull request:
Given that 3.6 is now several years old and the current and previous
 Ubuntu LTS ship with 3.8 or later, upgrade the minimum required Python
version from 3.6 to 3.8.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
doc: Require Python 3.8

Given that 3.6 is now several years old and the current and previous
 Ubuntu LTS ship with 3.8 or later, upgrade the minimum required Python
version from 3.6 to 3.8.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

mbolivar-nordic reviewed (commented):
I think it's time to update the GH workflows too.

mbolivar-nordic reviewed (changes requested):
In addition to the workflows, this is missing cmake/modules/python.cmake updates too, no? I think doc updates should come after we do all the actual work to move to the new version, not before

carlescufi commented:
> In addition to the workflows, this is missing cmake/modules/python.cmake updates too, no? I think doc updates should come after we do all the actual work to move to the new version, not before

Yep! Will push commits updating actions and cmake

mbolivar-nordic commented:
This should get a quick TSC ACK as well before merge, I think.

carlescufi committed:
actions: Make Python 3.8 the minimum required version

Change the matrix so that we now start testing 3.8+. No need to exclude
any particular OS/Python version combinations.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
cmake: Bump the minimum required Python version to 3.8

Given that 3.6 is now several years old and the current and previous
Ubuntu LTS ship with 3.8 or later, upgrade the minimum required Python
version from 3.6 to 3.8.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

tejlmand reviewed (dismissed):
lgtm

carlescufi commented on .github/workflows/devicetree_checks.yml:
```diff
@@ -20,13 +20,8 @@ jobs:
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8]
+        python-version: [3.8, 3.9, '3.10']
```
https://github.com/actions/setup-python/issues/160#issuecomment-724485470

mbolivar-nordic commented:
DNM to be super clear that this should get a TSC ACK before we move forward.

carlescufi commented:
No objections at the TSC
